codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -527,18 +538,26 @@ public ChannelFuture goAway(final ChannelHandlerContext ctx, final int lastStrea
 
             ChannelFuture future = frameWriter().writeGoAway(ctx, lastStreamId, errorCode, debugData, promise);
 
+            final String debugString = debugData.toString(UTF_8);
             future.addListener(new GenericFutureListener<ChannelFuture>() {
                 @Override
                 public void operationComplete(ChannelFuture future) throws Exception {
                     if (future.isSuccess()) {
                         if (errorCode != NO_ERROR.code()) {
+                            if (logger.isWarnEnabled()) {
+                                logger.warn(
```

Comment:
@nmittler - How about `debug` level?  This is the behavior defined by the http/2 spec, I'm not sure we should warn here?